src/dbus/dbus_request.hpp

```diff
@@ -43,39 +44,72 @@ class DBusRequest
         : mConnection(aConnection)
         , mMessage(aMessage)
     {
+        dbus_message_ref(aMessage);
```

Comment:
This seems to be really weird that `unref` is called automatically by the `std::unique_ptr` deleter. To do resource management automatically, we can defines `DBusConnectHolder` and `DBusMessageHolder` as a wrapper of the two `C` structs. Like this:
```C++
class DBusConnectionHolder {
public:
        DBusConnectionHolder(DBusConnection *aConnection) 
            : mConnection(aConnection) {
                assert(mConnection != nullptr);
                dbus_connection_ref(mConnection);
        }
     
        ~DBusConnectionHolder() {
                dbus_connection_unref(mConnection);
        }

        // TODO: copy/move constructor, assignment

        DBusConnection *get() { return mConnection; }

private:
        DBusConnection *mConnection;
};
```

Reply:
Or we just use raw `DBusConnection` and `DBusMessage` pointer, and `unref` them in dtor of `DBusRequest`.

Reply:
Updated